### ISSUE TITLE
impossible/error driver update Fedora 6.8.4-200

### ISSUE THREAD

**Issue by dakkar96** [morrownr/8814au#130]
Hi everyone, 

Sorry, big noob here. After upgrading to Fedora version "6.8.4-200.fc39.x86_64", my USB-WiFi dongle is not recognized anymore (TP-Link T9UH) .

I tried to re-install and it seemed to work fine in the Terminal since I have this message after writing "dkms status" command :

rtl8814au/5.8.5.1, 6.8.4-200.fc39.x86_64, x86_64: installed

But I still don't have wifi because my usb dongle is not recognized, even after multiple reboots ...
And when I jump back to an old Fedora version such as 6.7.11 everything is ok, my usb is recognized

Any idea how to solve this ? 

Many thanks in advance

**Comment by morrownr** [comment]
Run `sudo sh install-driver.sh` again and post the onscreen log here.

**Comment by dakkar96** [comment]
![Capture d’écran du 2024-04-08 09-44-46](https://github.com/morrownr/8814au/assets/127337370/37c5b26e-1b65-4706-bcf1-3369a543fa03)
![Capture d’écran du 2024-04-08 09-45-11](https://github.com/morrownr/8814au/assets/127337370/afdc54aa-8976-419e-8e9c-5cb404af623d)

I didn't make a reboot at this point for the screenshot and when I did it didn't work either.

**Comment by morrownr** [comment]
Reboot and then paste the output of:

$ dkms status

**Comment by dakkar96** [comment]
Ok I ran the remove-driver.sh script and then I processed a manual installation like this :
(in my driver folder)
$ make clean
$ make
$ sudo make install
$ sudo reboot

no errors during the process, but still no wifi after rebooting.

and weird thing, this is my dkms status right now :
![Capture d’écran du 2024-04-09 09-26-57](https://github.com/morrownr/8814au/assets/127337370/d7f24a7d-6793-4040-80ad-f2bb336467f4)

is this normal ? what should I do ?

**Comment by morrownr** [comment]
> is this normal ?

Yes, dkms status should show nothing since we ran remove-driver.sh.

> I ran the install-driver script. I still don't have wifi on this, but it came back for previous versions.

This is beginning to sound like a bug in fedora. Both methods of installation should have installed the driver to the currently active kernel. Since it is installing it to an older, inactive kernel, the question is why? You might want to present this info to a fedora forum to see what you get.

**Comment by dakkar96** [comment]
Everything came back to normal with update to 6.8.5-201

**Comment by morrownr** [comment]
That is good.

**Comment by morrownr** [comment]
@dakkar96 

Are you still stuck? If so, I have something I will merge so you can try it.

@morrownr

**Comment by dakkar96** [comment]
Hi, everything is working fine here with the driver. I am currently running Fedora 40 without issue. Thank you!

**Comment by morrownr** [comment]
Thanks for letting me know. If you running into problems as kernel upgrades flow into your system, let me know.

**Comment by dakkar96** [comment]
Hi there ! It seems that the driver doesn't work with Fedora 42 (6.14.3-300)

Any idea how to solve this ? 

Thanks in advance

**Comment by morrownr** [comment]
Hi @dakkar96 

I am in the process of discontinuing updates on the driver in this repo as we now have merged a new driver in the mainline Linux kernel. It went into kernel 6.15. What you can do is uninstall the driver in this repo:

$ sudo sh remove-driver.sh

...and then go to the following repo which is where we work on the new driver:

https://github.com/lwfinger/rtw88

Install that driver as it will work on your distro and when you get to kernel 6.15 or later, your adapter will be plug and play. WooHoo!

If you have problems with the rtw88 driver, open an issue in that repo.

Let me know how it goes.

**Comment by dakkar96** [comment]
everything works fine ! top project ! many thanks :)

**Comment by morrownr** [comment]
You are welcome. The devs that work on that driver watch issues in that rtw88 repo so if you have any problems, open an issue. See you over there.

**Comment by joseguzman1337** [comment]
Maintainer update: this issue has been addressed in upstream `main` at commit `b1866ce2b857a8dfe2e147e19eb8eca0a842ce18`. Resolution summary: DKMS staging cleanup + compiler-flag compatibility probing reduce kernel-update build failures in Fedora-like environments. This is mitigated with code and runtime validation tooling; if you still reproduce, please share fresh diagnostics from `tools/runtime-healthcheck.sh`.

**Comment by joseguzman1337** [comment]
Maintainer resolution update for #130: **impossible/error driver update Fedora 6.8.4-200**

Resolution applied in PR #3 / commit `b1866ce2b857a8dfe2e147e19eb8eca0a842ce18`: mitigated (DKMS staging cleanup + compiler-flag compatibility probing reduce kernel-update build failures in Fedora-like environments)
Commit: https://github.com/morrownr/8814au/commit/b1866ce2b857a8dfe2e147e19eb8eca0a842ce18
Implementation evidence:
- https://github.com/morrownr/8814au/blob/b1866ce2b857a8dfe2e147e19eb8eca0a842ce18/Makefile#L43

sk1 validation snapshot set (pre/post, no OS/kernel/driver reinstall/reboot):
- `/tmp/issue59-sk1-20260212-015134`

If this still reproduces, share exact repro steps + fresh runtime-healthcheck output from current main.
